Towers task place field

MRScheid opened the pull request:
Here's the first working version of the towers task place field.  I went the route of inheriting from the PlaceFieldWidget class and overwriting three of its methods.  Attached is a screen grab of the widget with the stub file.  I believe the firing rate is so low because it is a stub file that I'm using to generate the place field.  

![image](https://user-images.githubusercontent.com/10902625/101947896-2de97900-3bbf-11eb-9e68-552ced39864a.png)

Commit:
Towers task place field

First pass as towers task place field

Commit:
Revert "Towers task place field"

This reverts commit 360c17edb3d003fb9b59621aac6496534f60d64d.

Commit:
Towers task place field

First pass at place field for towers task

Commit:
Edit

Commit:
Fixed bug in accumulated evidence array calculation.

Commit:
First version of working towers task place field.

bendichter reviewed (changes requested):
@MRScheid it's hard to tell if this is working from the image. I expected a lot less white- it could be that you need to tweak some parameters or that the input data isn't really sufficient. Could you try to generate some fake data to get closer to what this should look like?

bendichter commented on tank_lab_to_nwb/towers_task_placefield.py:
```diff
@@ -0,0 +1,65 @@
+
+    def do_rate_map(self, index=0, speed_thresh=0.03, gaussian_sd_x=0.0184, gaussian_sd_y=0.0184, use_velocity=False):
+        occupancy, filtered_firing_rate, [edges_x, edges_y] = self.compute_twodim_firing_rate(index=index,
+                                                                                         speed_thresh=speed_thresh,
```
I don't think the speed_threshold is going to work here. I think it's going to compute speed on both dimensions of the "position"

bendichter commented on tank_lab_to_nwb/towers_task_placefield.py:
```diff
@@ -0,0 +1,65 @@
+
+    def get_controls(self):
+        style = {'description_width': 'initial'}
+        bft_gaussian_x = BoundedFloatText(value=0.0184, min=0, max=99999, description='gaussian sd (cm)', style=style)
+        bft_gaussian_y = BoundedFloatText(value=0, min=0, max=99999, description='gaussian sd (cm)', style=style)
```
The descriptions of these elements should reflect x vs y

bendichter commented on tank_lab_to_nwb/towers_task_placefield.py:
```diff
@@ -0,0 +1,65 @@
+        ax.set_ylim(np.nanmin(self.pos[:, 1] - 1), np.nanmax(self.pos[:, 1] + 2))
+        ax.set_xlim(0, np.nanmax(self.pos[:, 0] + 0.01))
+        cbar = plt.colorbar(im)
+        im.set_clim(0, np.nanmean(filtered_firing_rate))
```
Why is the clim max the mean of the data?

Commit:
Default value fixed, label fixed

Fixed default values for speed threshold and labels for x, y gaussian kernel.

Commit:
Overwrite pixel_width method

Overwriting the get_pixel_width in PlaceFieldWidget to set the pixel width to 1 for y (evidence) and to the appropriate value for x.

Commit:
Functional towers task place field widget

Fixed:
- Disable velocity button
- Fit rate map to plot area
- Fixed color scale or rate

Commit:
Added bin number control

Made modifications to add an additional control in the widget to control the number of bins in the towers task place field.
